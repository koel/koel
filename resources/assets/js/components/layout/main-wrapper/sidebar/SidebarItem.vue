<template>
  <li>
    <a :class="active && 'active'" :href="props.href">
      <icon :icon="props.icon" fixed-width />
      <span>
        <slot />
      </span>
    </a>
  </li>
</template>

<script lang="ts" setup>
import { Component, ref } from 'vue'
import  { useRouter } from '@/composables'

const props = defineProps<{ href: string; icon: Component, screen: ScreenName }>()

const active = ref(false)

const { onRouteChanged } = useRouter()

onRouteChanged(route => active.value = route.screen === props.screen)
</script>
