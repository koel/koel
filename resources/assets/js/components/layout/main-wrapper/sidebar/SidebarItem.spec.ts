import { expect, it } from 'vitest'
import UnitTestCase from '@/__tests__/UnitTestCase'
import { screen } from '@testing-library/vue'
import { faHome } from '@fortawesome/free-solid-svg-icons'
import SidebarItem from './SidebarItem.vue'

new class extends UnitTestCase {
  private renderComponent () {
    return this.render(SidebarItem, {
      props: {
        icon: faHome,
        href: '#',
        screen: 'Home'
      },
      slots: {
        default: 'Home'
      }
    })
  }

  protected test() {
    it('renders', () => expect(this.renderComponent().html()).toMatchSnapshot())

    it('activates when the screen matches', async () => {
      this.renderComponent()

      await this.router.activateRoute({
        screen: 'Home',
        path: '_',
      })

      expect(screen.getByRole('link').classList.contains('active')).toBe(true)
    })
  }
}
