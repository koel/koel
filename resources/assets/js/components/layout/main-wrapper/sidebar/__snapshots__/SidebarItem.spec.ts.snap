// Vitest Snapshot v1

exports[`renders 1`] = `<li><a class="" href="#"><br data-testid="icon" icon="[object Object]" fixed-width=""><span>Home</span></a></li>`;
